internal/worker.go

```diff
@@ -205,6 +205,10 @@ type (
 		// Optional: Sets opentracing Tracer that is to be used to emit tracing information
 		// default: no tracer - opentracing.NoopTracer
 		Tracer opentracing.Tracer
+
+		// Optional: Automatically send heartbeats for any activity with a HeartbeatTimeout
+		// default: false
+		ActivityAutoHeartbeat bool
```

Comment:
Having this option is fine for me

However it will be more ideal if there is another option also letting activity registration to disable autoHeartbeat for some activities.
Because people may want to have some with auto heartbeat,but some not. I believe 95% of the time people want autoHeartbeat, but some 5% they also want to do heartbeat in their code to control the heartbeat details.

Reply:
That makes sense, I had wondered if it would be better to have it set per activity instead of for the entire worker.  If this approach interferes with existing heartbeats that record details from the activity then it might be hard to use if it's enabled for all activities in the worker.  I can try to add that also.  Just to make sure I understand what that would look like:

* Leave this option here to make it possible to enable for all activities
* Add a new `DisableAutoHeartbeat` field to [RegisterActivityOptions](https://github.com/uber-go/cadence-client/blob/0256258b905b677f2f38fcacfbda43398d236309/internal/activity.go#L59).  This would probably need to be stored in `activityExecutor` after registration and exposed to `activityTaskHandlerImpl` by adding a new method to the internal `activity` interface.

Reply:
Yes, that's what I meant. 

If we are going this route, there are some more details I want to add:
* Probably rename `DisableAutoHeartbeat` to `DisableAutoHeartbeatIfWorkerHasEnabled`. Because `DisableAutoHeartbeat` sounds like it's already enabled by default. (or other better names if you can think of)
* This must be used only when AutoHeartbeat if enabled for worker, otherwise should return error
* Add clear comments on how to use it, and why this.

However, I feel that's quite confusing and error-prone. Because there are two options to control one behavior. Another route is that we only hanve `enableAutoHeartbeat` in RegisterActivityOptions. That will be much more clear for users. That will require every activity to use the option if they need autoHeartbeat, instead of a global option for the whole worker. But I believe the amount of code shouldn't be too much different -- as the option can be a global one and shared by all registration. 

I am leaning more tho the second approach, because it should be more friendly to users like you What do you think?

Reply:
I agree that it seems confusing to have an option in the worker and an overriding option when registering activities.  After giving it some more thought, it does feel like activity registration is the best place to have the option and not the worker.  It's the implementation of the activity that determines whether an auto-heartbeat is appropriate or not so it shouldn't depend on how the worker is configured.